add mf timespine behavior change (#6408)
fishtown-analytics/docs.getdbt.com: website/docs/reference/global-configs/behavior-changes.md

  - '{{ ... if flags.WHICH != 'freshness' }}'
```
</File>

### MetricFlow time spine YAML

> The flag is `False` by default.

Reviewer: ```suggestion
The `require_yaml_configuration_for_mf_time_spines` flag is set to `False` by default.
```

Author: good shout! I'll stick with the original and just add the flag name since your proposal is passive voice. so: 

The `require_yaml_configuration_for_mf_time_spines` flag is set to `False` by default.

> Set the `require_yaml_configuration_for_mf_time_spines` flag to `True` to require YAML configuration for MetricFlow time spine file for dbt Cloud Versionless or dbt Core 1.9 and later. In previous versions (dbt Core 1.8 and earlier), the MetricFlow time spine configuration was stored in a `metricflow_time_spine.sql` file.

Reviewer: ```suggestion
Set the flag to `True` to require YAML configuration for MetricFlow time spine file for dbt Cloud Versionless or dbt Core 1.9 and later. In previous versions (dbt Core 1.8 and earlier), the MetricFlow time spine configuration was stored in a `metricflow_time_spine.sql` file.
```

> Set the flag to `True` to require YAML configuration for MetricFlow time spine file for dbt Cloud Versionless or dbt Core 1.9 and later. In previous versions (dbt Core 1.8 and earlier), the MetricFlow time spine configuration was stored in a `metricflow_time_spine.sql` file.

Author: ```suggestion
In previous versions (dbt Core 1.8 and earlier), the MetricFlow time spine configuration was stored in a `metricflow_time_spine.sql` file.
```

When the flag is set to `True`, dbt will continue to support the SQL file configuration. When the flag is set to `False`, dbt will raise a deprecation warning if it detects a MetricFlow time spine configured in a SQL file. 

> The MetricFlow YAML file should be named `models/_models.yml` and should be located in the `models` directory.

Reviewer 2: I think the model can be named anything as long as it has the `time_spine:` field. Maybe we just link to the v1.9 docs [here](https://docs.getdbt.com/docs/build/metricflow-time-spine)?

> The MetricFlow YAML file should be named `models/_models.yml` and should be located in the `models` directory.

Author: ```suggestion
The MetricFlow YAML file should have the `time_spine:` field. Refer to [MetricFlow timespine](/docs/build/metricflow-time-spine) for more details. 
```

> When the flag is set to `True`, dbt will raise a deprecation warning if it detects a MetricFlow time spine configured a SQL file. When the flag is set to `False`, dbt will continue to support the SQL file configuration. 

Reviewer: ```suggestion
When the flag is set to `True`, dbt will raise a deprecation warning if it detects a MetricFlow time spine configured in a SQL file. When the flag is set to `False`, dbt will continue to support the SQL file configuration. 
```

> When the flag is set to `True`, dbt will raise a deprecation warning if it detects a MetricFlow time spine configured in a SQL file. When the flag is set to `False`, dbt will continue to support the SQL file configuration. 

Reviewer 2: Thanks for updating these docs, but this is the other way around. See my note in the PR description [here](https://github.com/dbt-labs/dbt-core/pull/10857)

> I added this to dbt_project.yaml in jaffle-sl-template and saw no warning for dbt parse:
flags:
  require_yaml_configuration_for_mf_time_spines: True

Author: oh right sorry, i missed the tests. thanks for clarifying @DevonFulcher ! so the flag allows users to continue to support the SQL file configuration. I'll update!

Author: updated it to this @DevonFulcher :

### MetricFlow time spine YAML
The `require_yaml_configuration_for_mf_time_spines` flag is set to `False` by default.

Set the flag to `True` to require YAML configuration for MetricFlow time spine file for dbt Cloud Versionless or dbt Core 1.9 and later. In previous versions (dbt Core 1.8 and earlier), the MetricFlow time spine configuration was stored in a `metricflow_time_spine.sql` file.

When the flag is set to `True`, dbt will continue to support the SQL file configuration. When the flag is set to `False`, dbt will raise a deprecation warning if it detects a MetricFlow time spine configured in a SQL file. 

The MetricFlow YAML file should be named `models/_models.yml` and should be located in the `models` directory.

Reviewer 2: Looks great! So, both configurations are accepted for now, regardless of the value of `require_yaml_configuration_for_mf_time_spines`, so this line isn't totally accurate:

> Set the flag to True to require YAML configuration for MetricFlow time spine file for dbt Cloud Versionless or dbt Core 1.9 and later.

I think we can just delete that sentence since you describe the True/False behavior later in this section.

Author: good flag, removed it now @DevonFulcher !
